bindings/matrix-sdk-ffi/src/timeline.rs

```diff
@@ -622,7 +628,19 @@ impl TryFrom<RumaMessageType> for MessageType {
                 },
             },
             RumaMessageType::Location(c) => MessageType::Location {
-                content: LocationContent { body: c.body, geo_uri: c.geo_uri },
+                content: LocationContent {
+                    body: c.body,
+                    geo_uri: c.geo_uri,
+                    description: c.location.clone().and_then(|l| l.description),
```

Comment:
The clone can be avoided here (we're getting the description here, and keeping the whole location alive because later we get the `zoom_level` out of it). I think you can avoid it by unpacking the `description` and `zoom_level` fields from the `c.location` at the same time, before defining this data structure:

```
let (description, zoom_level) = c.location.map(|loc| (loc.description, loc.zoom_level));
```

Reply:
Needs `.unzip()` at the end.

Reply:
Applied: https://github.com/matrix-org/matrix-rust-sdk/pull/2187/commits/64e82a2d78df5269c0ca60e523bed3e093f18937